Fix toolbar render problems in dark mode on resizing #536

de-jcup opened the pull request:
This PR
- closes #536

### Details
- Toolbar has in dark mode now a dedicated color instead of css inheritance. A parent container change by resizing does not change the toolbar color any longer
- The color is choosen in way that the main toolbar looks as expected before. Also the toggle border is now visible again, which could address #467 as well.

Signed-off-by: Albert Tregnaghi <albert.tregnaghi@gmail.com>

de-jcup committed:
Fix toolbar render problems in dark mode on resizing #536

- Toolbar has in dark mode now a dedicated color
  instead of css inheritance. A parent container
  change by resizing does not change the toolbar color any longer
- The color is choosen in way that the main toolbar looks as expected
  before. Also the toggle border is now visible again, which could
  address #467 as well.

Signed-off-by: Albert Tregnaghi <albert.tregnaghi@gmail.com>

de-jcup commented:
Here a screenshot from this commit and the chnged rendering 

![image](https://user-images.githubusercontent.com/11644753/215492993-d22acd8f-28ef-44b5-a5ef-f5f24746b49a.png)

iloveeclipse commented:
Just curious what is exact difference to the state before (I can't spot that)?

de-jcup commented:
@iloveeclipse : 
- for a screenshot with the problem please look at https://github.com/eclipse-platform/eclipse.platform.ui/issues/536#issuecomment-1408472509 
- or https://github.com/eclipse-platform/eclipse.platform.ui/issues/536#issue-1561996818

Target OS: Linux Mint 21.1, Native dark theme enabled (Mint-Y-Legacy-dark or any other dark theme)
 Tested with cinnamon (gnom3 based) + mate (gnome2 based): With both I have the mentioned problem

iloveeclipse commented:
Ok, if I understood it right, the difference is that the background of the toolbar changes and doesn't match the color of the toolbar buttons? But it should match them? For me both screenshots look inconsistent.

The "bad" case seem to show wrong background behind buttons in Console view but shows proper background in the front of them (note Package Explorer and Outline view). The "fixed" screenshot shows exact the opposite, which isn't better IMHO.

de-jcup committed:
Changed CSS background handling for ToolItem + CTabFolder #536

- removed CSS background handling for ToolItem + changed tests
  (temporary as long as there are conflicts)
- handled CTabFolder

de-jcup commented:
@iloveeclipse - thank you for the feedback, @akurtakov did already some test  about the PR and gave me feedback inside the issue itself.

@akurtakov : I really tried hard to fix the problems just with CSS but it did not work.

With last commit I reverted my former CSS attempts (by force push) and did changes inside code instead. It looks now good (see next picture). 

## Details
CTabFolder handling is now stopped, which stops the color changes when the toolbar menu is selected. 

I Also fixed the problem with the missing toggle states - the are shown now, but ... to bring up the toggle selection again and to fix the wrong background rendering for the tool items, I had to remove the CSS background handling for `ToolItem` which was introduced by @vogella with https://github.com/eclipse-platform/eclipse.platform.ui/commit/c07ab5389e1c9b75e46af9dd7cf0c522e35bba64
...  of course this not the correct way, but I was not able to find a way by CSS (and I really investigated much time)

Maybe this  could be a temporary solution/workaround? Currently as a dark mode user I am really unhappy with the rendering and the missing toggle selections.

![image](https://user-images.githubusercontent.com/11644753/216788915-064df752-d07f-4701-9beb-f864022439c2.png)

@vogella : Please take a look, maybe you have an Idea, how we could `ToolItem` css rendering work together with correct rendering inside CTabFolder.

vogella commented:
I think this is fixed with the recent changes of the dark theme by @mvm-sap. Please check again with the latest I-Build from https://download.eclipse.org/eclipse/downloads/
